iothub/service/samples/PnpServiceSamples/TemperatureController/Program.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.Devices;
+using Microsoft.Azure.Devices.Shared;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using PnpHelpers;
+
+namespace TemperatureController
+{
+    public class Program
+    {
+        // Get connection string and device id inputs
+        private static readonly string s_hubConnectionString = Environment.GetEnvironmentVariable("IOTHUB_CONNECTION_STRING");
+        private static readonly string s_deviceId = Environment.GetEnvironmentVariable("DEVICE_ID");
+
+        // These are values as defined in DTMI used for device client sample with component.
+        // DTDL interface used: https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/samples/TemperatureController.json
+        private const string ComponentName = "thermostat1";
+
+        // Writable property to update
+        private const string PropertyName = "targetTemperature";
+        private const double PropertyValue = 60;
+
+        // Method on a given component
+        private const string MethodName = "getMaxMinReport";
+        private static readonly DateTime dateTime = DateTime.Now ;
+
+        private static ServiceClient s_serviceClient;
+        private static RegistryManager s_registryManager;
+        private static ILogger s_logger;
+
+        public static async Task Main(string[] _)
+        {
+            ILoggerFactory loggerFactory = LoggerFactory.Create(builder =>
+            {
+                builder
+                .AddFilter(level => level >= LogLevel.Debug)
+                .AddConsole(options =>
+                {
+                    options.TimestampFormat = "[MM/dd/yyyy HH:mm:ss]";
+                });
+            });
+            s_logger = loggerFactory.CreateLogger<Program>();
+
+            await RunSampleAsync();
+        }
+
+        private static async Task RunSampleAsync()
+        {
+            s_logger.LogDebug($"Initialize the service client.");
+            InitializeServiceClient();
+
+            s_logger.LogDebug($"Get model Id and Update Component Property");
+            await GetAndUpdateTwinAsync();
+
+            s_logger.LogDebug($"Invoke a method on a Component");
+            await InvokeMethodAsync();
+        }
+
+        private static async Task InvokeMethodAsync()
+        {
+            // Create method name to invoke for component
```

Comment:
...for component `Thermostat`

Reply:
same as above. I prefer to keep it generic for comments